### PR TITLE
CRYPTO-127: CryptoInputStream#read should handle non-block case

### PR DESCRIPTION
**Pull request by kexianda**

### CODE REVIEW

**Comment by sundapeng** [comment]
+1

File: src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java [hunk 1]
```diff
@@ -255,8 +255,12 @@ public int read(byte[] b, int off, int len) throws IOException {
...
         // No data in the out buffer, try read new data and decrypt it
-        int nd = decryptMore();
-        if (nd <= 0) {
+        // we loop for new data
+        int nd = 0;
+        while (nd == 0) {
+            nd = decryptMore();
+        }
+        if (nd < 0) {
             return nd;
```
Comment (sundapeng) [inline src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java]: if we want to change to loop here, the return value should also be updated.

**Comment by sundapeng** [comment]
Thank Xianda, I have only one comment.

File: src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java (hunk 1 above)
Comment (kexianda) [inline src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java]: @sundapeng 
Hmm...,  it seems OK now. 
It's would be better if we add some testcases in future

Comment (sundapeng) [inline src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java]: Yes, I think it should be okay, I misunderstood it.

**Comment by sundapeng** [comment]
+1

**Comment by sundapeng** [comment]
Thank @kexianda

**Pull request closed without merging**